fuel-tx/src/tests/display.rs

```diff
@@ -32,6 +32,15 @@ fn to_from_str() {
 
             assert_eq!(tx, tx_p);
         });
+    TransactionFactory::<_, Upload>::from_seed(1295)
```

Comment:
I was gonna suggest maybe drying this up a little with a shared function:
```
fn transaction_round_trip_serde<T>()
    where 
        TransactionFactory<StdRng, T>: Iterator,
        T: Into<Transaction>
     {
    TransactionFactory::<_, Upgrade>::from_seed(1295)
        .take(20)
        .for_each(|(tx, _)| {
            // given
            let tx: Transaction = tx.into();
            // when
            let tx_p = tx.to_json();
            let tx_p = Transaction::from_json(tx_p).expect("failed to restore tx");
            // then
            assert_eq!(tx, tx_p);
}
```
I don't know if this is much better with all the generics.

I think the name `to_from_str` could be a little clearer either way though.